packages/client/src/views/VerifyCertificate/VerifyCertificatePage.tsx

```diff
@@ -254,17 +255,13 @@ export function VerifyCertificatePage() {
     const country =
       location?.address?.country &&
       intl.formatMessage(countryMessages[location?.address?.country])
-
     const city = location?.address?.city ?? EMPTY_STRING
-
     if (location?.address?.country === window.config.COUNTRY) {
-      const district =
-        location.address.district &&
-        offlineData.locations[location.address.district].name
-      const state =
-        location.address.state &&
-        offlineData.locations[location.address.state].name
-      return [city, district, state, country]
+      return [
+        city,
+        ...generateFullAddress(location.address, offlineData),
+        country
+      ]
         .filter((label) => Boolean(label))
         .join(', ')
```

Comment:
okay makes sense. We can then just create a `formatAddress` function which uses this `generateFullAdress` internally.

Reply:
although city and country will probably still be outside the `formatAddress` function as they are only specific to this page

Reply:
I would also need something for the `Location` as JPF mentioned there is value in seeing the full path in the `user.primaryOffice`. If you are not working on this, this can be merged and I can have a look for both of them.